charts/nr-k8s-otel-collector/templates/daemonset.yaml

```diff
@@ -0,0 +1,98 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ include "newrelic.common.naming.fullname" . }}-daemonset
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "newrelic.common.labels" . | nindent 4 }}
+spec: 
+  selector:
+    matchLabels:
+      {{- include "newrelic.common.labels.selectorLabels" . | nindent 6 }}
+      component: daemonset
+  template:
+    metadata:
+      labels:
+        {{- include "newrelic.common.labels.selectorLabels" . | nindent 8 }}
+        component: daemonset
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/daemonset-configmap.yaml") . | sha256sum }}
+    spec:
+      serviceAccountName: {{ include "newrelic.common.serviceAccount.name" . }}
+      containers: 
+        - name: otel-collector-daemonset
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          command:
+            - /otelcol-contrib
+            - --config=/conf/otel-collector-config.yaml
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
+          env:
+            - name: HOST_IP
+              valueFrom: 
+                fieldRef:
+                  fieldPath: status.hostIP 
+            - name: OTEL_EXPORTER_OTLP_ENDPOINT 
+              value: "http://$(HOST_IP):55680"
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: POD_UID
+              valueFrom: 
+                fieldRef:
+                  fieldPath: metadata.uid
+            - name: KUBE_NODE_NAME
+              valueFrom:
+                fieldRef:
+                  apiVersion: v1
+                  fieldPath: spec.nodeName
+            - name: OTEL_RESOURCE_ATTRIBUTES
+              value: service.instance.id=$(POD_NAME),k8s.pod.uid=$(POD_UID)
+            - name: NR_LICENSE_KEY
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "newrelic.common.license.secretName" . }}
+                  key: {{ include "newrelic.common.license.secretKeyName" . }}
+          volumeMounts:
+            # TODO (chris): Comment out below when not linux
+            - name: host-fs
+              mountPath: /hostfs
+              readOnly: true
+            - name: config-volume
+              mountPath: /conf
+            - name: varlogpods
+              mountPath: /var/log/pods
+              readOnly: true
+            - name: varlibdockercontainers
+              mountPath: /var/lib/docker/containers
+              readOnly: true
+      volumes:
+        # TODO (chris): Comment out below when not linux
+        - name: host-fs
+          hostPath:
+            path: /
```

Comment:
There is no way to "comment out" something depending on the OS.

You'll need two daemonsets having different selectors:
https://kubernetes.io/docs/reference/labels-annotations-taints/#kubernetes-io-os

Reply:
Yeah, sorry this is just a reminder to myself that this won't work on non linux nodes, but thanks for the headsup on creating a new daemonset to support that functionality. that makes sense. we won't be including non linux nodes support in this PR, and i'll remove my comments since I've noted them down elsewhere, and we will be including the limitations of this agent in documentation otherwise.

Reply:
Awesome :)